### ISSUE TITLE
DelimitedFiles should now require compat entries

### ISSUE THREAD

**Issue by KristofferC** [JuliaRegistries/RegistryCI.jl#471]

**Comment by KristofferC** [comment]
Yes, RegistratorCI is clearly where this should be opened. I moved it.

**Comment by ericphanson** [comment]
Taking another look at this:

https://github.com/JuliaRegistries/RegistryCI.jl/blob/0c50828db74921c7cd34b18da47ac0ddf9376a39/src/AutoMerge/guidelines.jl#L91

https://github.com/JuliaRegistries/RegistryCI.jl/blob/5f0aba4f6b95bfb63f03ab803f39721c2cd55baf/src/AutoMerge/util.jl#L232-L244

https://github.com/JuliaRegistries/General/blob/a71c6fbf0766c12f54ea1cf23a3e7dc7277f380a/.github/workflows/automerge.yml#L22

It looks like once 1.9 is out, General's AutoMerge workflow will switch to 1.9, and presumably those Pkg internals will no longer have DelimitedFiles as a stdlib, so we will start requiring compat then.

Does that sound reasonable? Do we want to edit it so we rm DelimitedFiles from that list early so we start requiring it already?
